File: gaia-sdk-python/gaia_sdk/graphql/request/input/CreateBehaviourImpulse.py


from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class CreateBehaviourImpulse():
    """
    The specification to create a behaviour instance
    """
    identityId: str
    qualifier: str
    appendent: str
    behaviour: str
    labelList: List[str]

    def __init__(self, identityId: str, qualifier: str, appendent: str, behaviour: str, labelList: List[str]):
        self.identityId = identityId
        self.qualifier = qualifier
        self.appendent = appendent
        self.behaviour = behaviour
        self.labelList = labelList

    def __eq__(self, other):
        if type(other) is type(self):
            return self.identityId == other.identityId and self.qualifier == other.qualifier and self.appendent == other.appendent and self.behaviour == other.behaviour and self.labelList == other.labelList
        return False

    def __repr__(self):
        return {'identityId': self.identityId, 'qualifier': self.qualifier, 'appendent': self.appendent, 'behaviour': self.behaviour, 'labelList': self.labelList}

File: gaia-sdk-javascript/src/graphql/request/input/CreateBehaviourImpulse.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../enumeration/RuntimeState";
import {SkillState} from "../enumeration/SkillState";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * The specification to create a behaviour instance
 */
export class CreateBehaviourImpulse {
public _typeName = "CreateBehaviourImpulse";
    private identityId:Uuid;
    private qualifier:String;
    private appendent:String;
    private behaviour:String;
    private labelList:Array<String>;

    constructor (identityId:Uuid, qualifier:String, appendent:String, behaviour:String, labelList:Array<String>) {
        this.identityId = identityId;
        this.qualifier = qualifier;
        this.appendent = appendent;
        this.behaviour = behaviour;
        this.labelList = labelList;
    }
}
